codice/ddf
# [2.19.x] G-9577 Display error banner for export KML with no location

ported ddf-ui PRs:
3.4.x codice/ddf-ui#511

ported ddf PRs:
2.23.x https://github.com/codice/ddf/pull/6511
master codice/ddf#6514

---------------------
#### What does this PR do?
This PR addresses following issues -
Main Issue:
1.  When user tries to export a format that requires "location" attribute, they are left with nothing occurring (e.g. KML/KMZ).
**Fixed issue new behavior:**  If a Single result is selected to export, the option for the export type they do not qualify for will not be displayed (e.g. a result that has no location, will not show KML/KMZ as an export option. Note - this behavior is not true for table export, it always shows all available export options). If multiple results are selected and some has the required attribute and some do not, the file will export the one(s) with the required attribute and will show a warning banner with the id(s) of the ones that were not exported. For Table export if try to only export result(s) that do not have any location attribute, an error dialogue will be displayed, saying the missing attribute was not provided and nothing is exported.

Other issues:
1.  For table export, when a filter is applied to the results, and user select "Visible Results", exported item does not match Results that are actually present after the filter was applied.
2. ~~For the export in Inspector->Actions tab, CVS was not exporting.~~ resolved in https://github.com/codice/ddf/pull/6503
#### Who is reviewing it? 
@andrewzimmer 
@hayleynorton
@zta6 
@Lambeaux 
@mojogitoverhere 
@millerw8 
@cassandrabailey293



#### Select relevant component teams: 


#### Ask 2 committers to review/merge the PR and tag them here.


#### How should this be tested?

• build / run / profile install / upload sample data
• Execute a search and either add or ensure at least one or some of the result(s) have a "location" attribute.

• For main issue 1: Test the different types of exports and ensure behaves as described in Fixed issue new behavior. Also ensure no regression for https://github.com/codice/ddf/pull/6496
1.  Table export (Note - All export option(s) are always shown even if the item to export does not have required attribute, it gets caught in back end, and responds with appropriate message)
		a. Via the "Export" button top far right. Can choose 3 different set of exports ("Visible Results", "All Results", "Specific Number of Results") for specific export type.
2.  All other exports (Note - These types of export only display options for items that has its required attributes present. e.g. KML/KMZ will only appear as option for result(s) that has  a "location attribute")
		a. Via an individual Result lower right three-dot menu ("Export as")
		b. Via three-dot menu on the search pane (has 2 types of exports: "Export Selected" and "Export Selected(Compressed)"). Also note, can select multiple result(s) to export.
		c. Via Inspector tab three-dot menu (can select single and/or multiple result(s))
		d. Via Inspector tab, Actions subtab (Actions subtab only appear when 1 Result is selected)

• For other issues 1:
1.   Filter the results you have so there is less item displayed and others are 
2.  Do a table export, with selecting "Visible Results", and can choose CSV or other format and clicking download
3.  Ensure only the visible results in table after filter was applied are visible
	
#### Any background context you want to provide?

#### What are the relevant tickets?
Fixes: #____

#### Screenshots


https://user-images.githubusercontent.com/65194214/106331988-1e8fbf00-6243-11eb-8f44-56bc6c213923.mp4


#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
